src/fitnesse/components/ComponentFactory.java

```diff
@@ -76,12 +82,26 @@ public ComponentFactory(Properties properties) {
     }
   }
 
+  public <T> T createComponentForClassName(String className) {
+    Class<T> clazz;
+    try {
+      clazz = lookupComponentClass(className);
+    } catch (ClassNotFoundException e) {
+      throw new ComponentInstantiationException("Unable to load class named " + className, e);
+    }
+    return createComponent(clazz);
+  }
+
+  public <T> Class<T> lookupComponentClass(String className) throws ClassNotFoundException {
+    return (Class<T>) classLoader.loadClass(className);
```

Comment:
Should this not be delegated to `ClassUtil` also? It seems a bit strange that the class loader is maintained both in that util singleton and this class. It would be nicer if all class loader access is done in a uniform way.

Reply:
As said above, I'd like to get rid of `ClassUtils` altogether.

Reply:
Maybe the static method can be moved to `ComponentFactory`, removing `ClassUtils`? Because it seems that already has the responsibility to load stuff.

Reply:
That would not solve the problem of having *mutable* static variables. The fact that we need those points to a design flaw.